lua/lspconfig/ui/lspinfo.lua

```diff
@@ -64,9 +64,13 @@ local function make_config_info(config, bufnr)
   local buffer_dir = api.nvim_buf_call(bufnr, function()
     return vim.fn.expand '%:p:h'
   end)
-  local root_dir = config.get_root_dir and config.get_root_dir(buffer_dir)
-  if root_dir then
-    config_info.root_dir = root_dir
+
+  if config.get_root_dir then
+    local status, root_dir = coroutine.resume(coroutine.create(vim.schedule_wrap(function()
+      local root_dir = config.get_root_dir(buffer_dir)
+      coroutine.yield(root_dir)
+    end)))
+    config_info.root_dir = status and root_dir or error_messages.root_dir_not_found
```

Comment:
Let's not use `coroutine.yield` outside of the "user code" and pass the result through a reference/upvalue. If for some reason the initial suspend is something like `coroutine.yield(42)`, `root_dir` will be assigned to 42. Not that anyone should do that, but a reference will be more robust.
```suggestion
    local root_dir
    local status = coroutine.resume(coroutine.create(vim.schedule_wrap(function()
      root_dir = config.get_root_dir(buffer_dir)
    end)))
    config_info.root_dir = status and root_dir or error_messages.root_dir_not_found
```
But doing this alone will make it stop working for anything has an async `root_dir` function. Preferably the buffer could be updated with this information once we actually get the result. Maybe checking if `coroutine.status` is `'suspended'`, and if so then save the line number and update that line? But this will make the code more complicated.

Reply:
> Maybe checking if coroutine.status is 'suspended', and if so then save the line number and update that line? But this will make the code more complicated.

honestly yes .. I don't have good idea on lspinfo .  It has some importance. But not that important.